packages/atomic/src/components/ipx/atomic-ipx-button/atomic-ipx-button.tsx

```diff
@@ -0,0 +1,102 @@
+import {Component, Element, h, Prop} from '@stencil/core';
+import CloseIcon from 'coveo-styleguide/resources/icons/svg/close.svg';
+import SearchIcon from 'coveo-styleguide/resources/icons/svg/search.svg';
+import {Button} from '../../common/button';
+
+/**
+ * @internal
+ */
+@Component({
+  tag: 'atomic-ipx-button',
+  styleUrl: './atomic-ipx-button.pcss',
+  shadow: true,
+})
+export class AtomicIPXButton {
+  @Element() public host!: HTMLElement;
+
+  /**
+   * The label that will be shown to the user.
+   */
+  @Prop({reflect: true}) public label = 'Workplace Search';
+
+  /**
+   * Whether the IPX layout is open.
+   */
+  private isOpen = false;
+
+  private async onClick() {
+    this.isOpen ? this.close() : this.open();
+    this.isOpen = !this.isOpen;
+
+    this.render();
+  }
+
+  private renderIPXButton() {
+    return (
+      <Button
+        style="primary"
+        part="ipx-button"
+        class="my-2 btn"
+        onClick={() => this.onClick()}
+      >
+        <span class="btn-icon" part="button-icon">
+          <atomic-icon
+            class="svg-open"
+            icon={this.getIcon(CloseIcon)}
+          ></atomic-icon>
+          <atomic-icon
+            class="svg-close"
+            icon={this.getIcon(SearchIcon)}
+          ></atomic-icon>
+        </span>
+        <span class="buttonText" part="button-text">
+          {this.label}
+        </span>
+      </Button>
+    );
+  }
+
+  public render() {
+    return (
+      <div class="flex flex-col items-center" part="container">
+        {this.renderIPXButton()}
+      </div>
+    );
+  }
+
+  private get button() {
+    return this.host.shadowRoot!.querySelector('button');
+  }
+
+  private get ipxLayout() {
+    return document.querySelector('atomic-ipx-layout')!;
+  }
+
+  private open() {
+    this.ipxLayout.style.height = `${Math.min(
+      540,
+      document.documentElement.clientHeight
+    )}px`;
+    this.ipxLayout.style.visibility = 'visible';
+    this.ipxLayout.style.pointerEvents = 'initial';
+    this.ipxLayout.style.opacity = '1';
```

Comment:
Instead of modifying the `style` directly, I recommend making a component that wraps the contents of the IPX interface (similar to `atomic-modal`, and this component would live inside `atomic-search-interface`) and making this component render an `atomic-focus-trap` to ensure that the focus can't exit IPS while it's open and to ensure that screen readers can't access the interface while it's closed.

Reply:
I'll have a look at the modal component but i don't think we want to trap focus on IPX. 
The purpose is to let users used it while doing other things on a page iirc.

Reply:
In discussion with @adroletCoveo and the team.

Reply:
@btaillon button was validated by Audrey. I added an IPX modal component with focus trap.